serve/mlc_serve/model/base.py

```diff
@@ -0,0 +1,45 @@
+from dataclasses import dataclass
+from typing import Optional
+import os 
+import json
+import inspect
+
+@dataclass
+class ModelArtifactConfig:
+    model_artifact_path: Optional[str] = None
+    num_shards: Optional[int] = None
+    quantization: Optional[str] = None
+    model_type: Optional[str] = None
+    library_name: Optional[str] = None
+    max_context_length: Optional[int] = None
+    vocab_size: Optional[int] = None
+    sliding_window: Optional[int] = None
+    build_options: Optional[str] = None
+    num_key_value_heads: Optional[int] = None
+    num_attention_heads: Optional[int] = None
+    num_hidden_layers: Optional[int] = None
+    hidden_size: Optional[int] = None
+
+    @classmethod
+    def _from_json(config_cls, json_obj: dict):
```

Comment:
Just as a future reference using something like Pydantic actually might be better than dataclasses as you get validation + json conversion with error handling for free. We can always put a follow up ticket here.

Reply:
Is there any example? Haven't used Pydantic before.

Reply:
add this as TODO item